File: ios/ModelFace.swift
//
//  ModelFace.swift
//  iosTensorflowLite
//
//  Created by Yudi Edri Alviska on 06/08/21.
//

import Foundation
import UIKit
import TensorFlowLite

class ModelFace {
    var id: String?
    var title: String?
    var distance: Float?
    var extra: Tensor?
    var location: CGRect?
    var color: UIColor?
    var crop: UIImage?
    
    init(id: String, title: String, distance: Float, location: CGRect) {
        self.id = id
        self.title = title
        self.distance = distance
        self.location = location
    }
    
    func setId(id: String) {
        self.id = id
    }
    
    func setTitle(title: String) {
        self.title = title
    }
    
    func setDistance(distance: Float) {
        self.distance = distance
    }
    
    func setExtra(extra: Tensor) {
        self.extra = extra
    }
    
    func setLocation(location: CGRect) {
        self.location = location
    }
    
    func setColor(color: UIColor) {
        self.color = color
    }
    
    func setCrop(crop: UIImage) {
        self.crop = crop
    }
    
    func getId() -> String? {
        return self.id
    }
    
    func getTitle() -> String? {
        return self.title
    }
    
    func getDistance() -> Float? {
        return self.distance
    }
    
    func getExtra() -> Tensor? {
        return self.extra
    }
    
    func getLocation() -> CGRect? {
        return self.location
    }
    
    func getColor() -> UIColor? {
        return self.color
    }
    
    func getCrop() -> UIImage? {
        return self.crop
    }
}

File: android/src/main/java/com/reactnativefacerecognition/recognition/ModelFace.java
package com.reactnativefacerecognition.recognition;

import android.graphics.RectF;

public class ModelFace {
  private final String id;
  private final String title;
  private final Float distance;
  private Object extra;
  private RectF location;
  private Integer color;

  public ModelFace(final String id, final String title, final Float distance, final RectF location) {
    this.id = id;
    this.title = title;
    this.distance = distance;
    this.location = location;
  }

  public String getTitle() {
    return title;
  }

  public Float getDistance() {
    return distance;
  }

  public Object getExtra() {
    return extra;
  }

  public void setExtra(Object extra) {
    this.extra = extra;
  }

  public RectF getLocation() {
    return location;
  }

  public void setLocation(RectF location) {
    this.location = location;
  }

  public Integer getColor() {
    return color;
  }

  public void setColor(Integer color) {
    this.color = color;
  }

  @Override
  public String toString() {
    return "ModelRecognition{" +
      "id='" + id + '\'' +
      ", title='" + title + '\'' +
      ", distance=" + distance +
      ", extra=" + extra +
      ", location=" + location +
      ", color=" + color +
      '}';
  }
}
